File: deepgoal/core/types/tools.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel, Field

from .enums import McpServerType


class McpServerConfig(BaseModel):
    type: McpServerType = McpServerType.STDIO
    command: str | None = None
    args: list[str] = Field(default_factory=list)
    url: str | None = None
    headers: dict[str, str] = Field(default_factory=dict)
    env: dict[str, str] = Field(default_factory=dict)


class ToolConfig(BaseModel):
    name: str
    params: dict[str, Any] = Field(default_factory=dict)


class SkillConfig(BaseModel):
    name: str
    args: str = ""

File: deepgoal-ts/src/core/types/tools.ts
import type { JsonObject } from './common.js';
import { McpServerType } from './enums.js';

export interface ToolParams {
  values: JsonObject;
}

export interface McpServerConfig {
  type: McpServerType;
  command?: string | undefined;
  args: string[];
  url?: string | undefined;
  headers: Record<string, string>;
  env: Record<string, string>;
}

export interface ToolConfig {
  name: string;
  params: ToolParams;
}

export interface SkillConfig {
  name: string;
  args: string;
}

export function createToolParams(values: JsonObject = {}): ToolParams {
  return { values };
}

export function createMcpServerConfig(config: Partial<McpServerConfig> = {}): McpServerConfig {
  return {
    type: config.type ?? McpServerType.Stdio,
    command: config.command,
    args: config.args ?? [],
    url: config.url,
    headers: config.headers ?? {},
    env: config.env ?? {},
  };
}

export function createToolConfig(name: string, params: ToolParams = createToolParams()): ToolConfig {
  return { name, params };
}

export function createSkillConfig(name: string, args = ''): SkillConfig {
  return { name, args };
}
